### ISSUE TITLE
Allow Count Up Block to Pull in Data

### ISSUE THREAD

**Issue by stevejburge** [publishpress/PublishPress-Blocks#520]
Could the Count Up Block automatically pull in data from the site?

For example:
- Number of users
- Number of posts
- Number of comments
- Number of WooCommerce or EDD purchases

Could be a Pro featuree

**Comment by htmgarcia** [comment]
@stevejburge this feature sounds promising! By the moment the content is static. We'll check and see how to implement queries from database.

**Comment by contactashish13** [comment]
@htmgarcia please let me know if you want me to start working on this.

**Comment by htmgarcia** [comment]
@contactashish13 please give a try.

**Comment by contactashish13** [comment]
@htmgarcia we can embed a [codemirror](https://codemirror.net/) editor in the inspector for custom queries (for pro version). For the free version we can give a dropdown with multiple options viz. number of users, number of posts etc. where each option corresponds to a prebuilt query. The pro version can thus have additional prebuilt queries (e.g. for woocommerce or EDD) + the option to write custom queries.

Let me know what you think.

**Comment by htmgarcia** [comment]
@contactashish13 I would suggest this as a Pro feature only. The dropdown seems a great approach due is good for UX.

**Comment by htmgarcia** [comment]
@contactashish13 any luck with this one?

**Comment by contactashish13** [comment]
@htmgarcia I started on this last week but left it midway. Will look at it this week :)

**Comment by contactashish13** [comment]
@htmgarcia this is how it will behave. The video doesn't show the hover behavior - when you hover on `Dynamic` it shows the query (in this case `# Posts` or `# Pages`) as a HTML `title` so that the user is aware what query it corresponds to.

https://user-images.githubusercontent.com/12953439/126942153-8e4030d6-d1a4-429b-82a8-7cbf3d76e07a.mp4

I've added only 2 queries but more can be added by us (or by the user as a filter is provided). Let me know if you foresee any changes

PS - I've added the code to the free plugin for my testing and will migrate the code to pro.

**Comment by htmgarcia** [comment]
@contactashish13 the approach looks promising! Let's consider these queries, please:
- Number of users
- Number of posts (by post type: post, page and WooCommerce products)
- Number of comments
- Number of WooCommerce or EDD purchases

Having this feature in Pro is possible without duplicating the Count Up block code from Free?

**Comment by contactashish13** [comment]
@htmgarcia it is possible to support that if both free and pro are activated. But from what I remember only one of them should be kept active, right?

On Jul 26, 2021, 22:16, at 22:16, "Valentín" ***@***.***> wrote:
***@***.*** the approach looks promising! Let's consider these

**Comment by htmgarcia** [comment]
@contactashish13 both plugins can be active however that is not mandatory. Is important to mention that when building the zip for Pro, the files from Free are added to Pro zip, so having only Pro version is enough.

**Comment by contactashish13** [comment]
@htmgarcia then what do you suggest? Should I keep the entire code only in pro or have a mix where only the dynamic queries part is in pro and the rest of the code is in free?

On Jul 27, 2021, 19:54, at 19:54, "Valentín" ***@***.***> wrote:
***@***.*** both plugins can be active however that is not

**Comment by htmgarcia** [comment]
@contactashish13 new plan. Let's hold on this issue and give priority to these following the same order, please
#646
#647
#441

**Comment by contactashish13** [comment]
@htmgarcia let me know if I can start working on this in parallel

**Comment by contactashish13** [comment]
@htmgarcia let me know when I can restart this. The code is going stale :)

**Comment by htmgarcia** [comment]
@contactashish13 let's hold on this one, please.
